examples/spotify-api-example/src/main/java/com/spotify/apollo/example/AlbumResource.java

```diff
@@ -0,0 +1,113 @@
+package com.spotify.apollo.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.spotify.apollo.Client;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.RequestContext;
+import com.spotify.apollo.Response;
+import com.spotify.apollo.example.data.Album;
+import com.spotify.apollo.example.data.Artist;
+import com.spotify.apollo.route.AsyncHandler;
+import com.spotify.apollo.route.JsonSerializerMiddlewares;
+import com.spotify.apollo.route.Route;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import okio.ByteString;
+
+/**
+ * The album resource demonstrates how to use asynchronous routes in Apollo.
+ */
+public class AlbumResource {
+
+  private static final String SPOTIFY_API = "https://api.spotify.com";
+  private static final String SEARCH_API = SPOTIFY_API + "/v1/search";
+  private static final String ALBUM_API = SPOTIFY_API + "/v1/albums";
+
+  private final ObjectMapper objectMapper;
+  private final ObjectWriter objectWriter;
+
+  public AlbumResource(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+    this.objectWriter = objectMapper.writer();
+  }
+
+  public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
+    // The album resource has two routes. Since both are returning the same type,
+    // we can map them to the same middleware
+    // Note that this could also have been set up as a single route to "/albums/<tag>".
+    return Stream.of(
+        Route.async("GET", "/albums/new", context -> getAlbums(context, "new")),
+        Route.async("GET", "/albums/hipster", context -> getAlbums(context, "hipster"))
+    )
+        .map(route -> route.withMiddleware(
+            JsonSerializerMiddlewares.jsonSerializeResponse(objectWriter)));
+  }
+
+  public CompletionStage<Response<ArrayList<Album>>> getAlbums(RequestContext requestContext, String tag) {
```

Comment:
This could have lower accessibility, since it's only going to be called internally from this class. I think package-private would make sense, so that it could be unit tested.